### ISSUE TITLE
Alternative deployment method minor issues

### ISSUE THREAD

**Issue by romanini-ciandt** [terraform-google-modules/terraform-example-foundation#1067]
### TL;DR

Some of the new alternative deployment methods (GitHub and Terraform Cloud) contains minors adjusts that need to be done in order to deploy the whole foundation smoothly.

### Expected behavior

Successful run just by following all the GitHub or Terraform Cloud READMEs 

### Observed behavior

Related to GitHub alternative deployment :
- When the output of the GitHub action linked to opened PRs is too large (more than 65K characters) the output preview can't fit at the PR's page and causes an error. (See more details in this similar bug: https://github.com/actions/github-script/issues/266);
- Errors related to missing `self_impersonate` permission for SAs;

Related to Terraform Cloud alternative deployment :
- Missing an index (`e.g: [index]`) in a tfe_agent_pool.tfc_agent_pool resource usage;

### Terraform Configuration

```hcl
N/A
```

### Terraform Version

```sh
N/A
```
